code/modules/vehicles/vehicle_actions.dm

```diff
@@ -175,4 +175,19 @@
 		var/obj/vehicle/sealed/car/clowncar/C = vehicle_entered_target
 		if(C.cannonbusy)
 			to_chat(owner, "<span class='notice'>Please wait for the vehicle to finish its current action first.</span>")
-		C.ToggleCannon()
\ No newline at end of file
+		C.ToggleCannon()
+
+/datum/action/vehicle/sealed/Thank
+	name = "Thank the Clown car Driver"
+	desc = "They're just doing their job."
+	button_icon_state = "car_thanktheclown"
+	var/last_thank_time
+
+/datum/action/vehicle/sealed/Thank/Trigger()
+	if(istype(vehicle_entered_target, /obj/vehicle/sealed/car/clowncar))
+		var/obj/vehicle/sealed/car/clowncar/C = vehicle_entered_target
+		if(world.time >= last_thank_time + 60)
+			var/mob/living/carbon/human/clown = pick(C.return_drivers())
+			owner.say("Thank you for the fun ride [clown.name]!")

```

Comment:
Shouldn't there be a comma between "ride" and the driver's name?

Reply:
yes there should be